server/src/services/mindmap.service.ts

```diff
@@ -0,0 +1,95 @@
+import { PrismaClient, Mindmap, Node, Connection } from '@prisma/client';
+import { prisma as prismaService } from './prisma.service';
+import { BaseDatabaseService } from './base-database.service';
+
+export interface MindmapCreateInput {
+  name: string;
+  userId: number;
+}
+
+export class MindmapService extends BaseDatabaseService<Mindmap> {
+  constructor(protected readonly prisma: PrismaClient = prismaService) {
+    super(prisma);
+  }
+
+  async find(id: number): Promise<Mindmap | null> {
+    return await this.prisma.mindmap.findUnique({
+      where: {
+        id,
+      },
+      include: {
+        Node: true,
+        Connection: true,
+      }
+    });
+  }
+
+  async findOrThrow(id: number): Promise<Mindmap> {
+    const mindmap = await this.find(id);
+
+    if (!mindmap) {
+      throw new Error(`Mindmap with id ${id} not found`);
+    }
+
+    return mindmap;
+  }
+
+  async findManyByIds(ids: number[]): Promise<Mindmap[]> {
+    return await this.prisma.mindmap.findMany({
+      where: {
+        id: {
+          in: ids,
+        },
+      },
+    });
+  }
+
+  async getNodes(id: number): Promise<Node[]> {
```

Comment:
No need to overcomplicate stuff, we can just call the find by id and the connection or nodes. There is no need of these methods. They do not optimize anything, just add more boileplate